lib/core/src/Cardano/Wallet.hs

```diff
@@ -615,6 +618,25 @@ restoreWallet ctx wid = db & \DBLayer{..} -> do
     run :: ExceptT ErrNoSuchWallet IO () -> IO (FollowAction ErrNoSuchWallet)
     run = fmap (either ExitWith (const Continue)) . runExceptT
 
+-- | Rewind the UTxO snapshots, transaction history and other information to a
+-- the earliest point in the past that is before or is the point of rollback.
+rollbackBlocks
+    :: forall ctx s k.
+        ( HasLogger ctx
+        , HasDBLayer s k ctx
+        )
+    => ctx
+    -> WalletId
+    -> SlotId
+    -> ExceptT ErrNoSuchWallet IO ()
+rollbackBlocks ctx wid point = db & \DBLayer{..} -> do
+    liftIO $ logInfo tr $ "Try rolling back to " <> pretty point
+    point' <- mapExceptT atomically $ rollbackTo (PrimaryKey wid) point
+    liftIO $ logInfo tr $ "Rolled back to " <> pretty point'
```

Comment:
Nice logs!

Reply:
Yeah, these tree lines make it somewhat obvious what (one of) the problem was. Took me long enough though...